### PR TITLE
[CAR] Controller, Service and Mapper

### PR DESCRIPTION
**Pull request by pastcwdt**

### CODE REVIEW

File: src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CarServiceImpl implements CarService {
+    private final CarRepository carRepository;
+    private final CarMapper carMapper;
+
+    @Override
+    @Transactional
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java:22]: You don't need Transactional here. You interact with DB only with save method
Check if it works without it

File: src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    public CarDto save(CreateCarRequestDto request) {
+        Car car = carMapper.toEntity(request);
+        car.setType(Car.CarType.valueOf(request.type()));
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    @Transactional
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java:30]: Same here. Check if works without Transactional

File: src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    public List<CarDto> getAll() {
+        return carRepository.findAll().stream()
+                .map(carMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java:38]: Same here. Check if works without Transactional
You don't interact with DB here

File: src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    public CarDto getById(Long id) {
+        return carMapper.toDto(carById(id));
+    }
+
+    @Override
+    @Transactional
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java:44]: Same

File: src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    public CarDto update(Long id, CreateCarRequestDto request) {
+        Car car = carById(id);
+        updateValues(car, request);
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java:47]: This should be done with mapper

File: src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        );
+    }
+
+    private void updateValues(Car car, CreateCarRequestDto request) {
+        car.setModel(request.model());
+        car.setBrand(request.brand());
+        car.setType(Car.CarType.valueOf(request.type()));
+        car.setInventory(request.amountAvailable());
+        car.setDailyFee(request.dailyFee());
+    }
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java:72]: Better delete this mapper and add method to mapper: 
`void updateCar(CreateCarRequestDto dto, @MappingTarget Car car);`
and use it in update method in service

File: src/main/java/com/champions/carsharingservice/controller/CarController.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    private final CarService carService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public CarDto add(@RequestBody @Valid CreateCarRequestDto request) {
+        return carService.save(request);
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/controller/CarController.java:33]: It is default. You may remove it

File: src/main/java/com/champions/carsharingservice/controller/CarController.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    public List<CarDto> getAll() {
+        return carService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public CarDto getById(@PathVariable Long id) {
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/controller/CarController.java:40]: Lets put @Positive annotation before Long id and @Validaded annotation above the controller

File: src/main/java/com/champions/carsharingservice/controller/CarController.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        return carService.getById(id);
+    }
+
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public CarDto update(@PathVariable Long id, @RequestBody @Valid CreateCarRequestDto request) {
```
Comment (TetianaDenchyk) [inline src/main/java/com/champions/carsharingservice/controller/CarController.java:46]: Same about responsestatus and @Positive

**Review by TetianaDenchyk — CHANGES_REQUESTED** [review]
Good job! :)
Let's discuss and fix 
Update your branch, you will have security in it so that you can add **access by roles**
Add **swagger** also please

File: src/main/java/com/champions/carsharingservice/controller/CarController.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    @Operation(summary = "Add a new car", description = "Add a new car")
+    public CarDto add(@RequestBody @Valid CreateCarRequestDto request) {
+        return carService.save(request);
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Search cars",
+            description = "Get list of cars with specified parameters")
+    public List<CarDto> getAll() {
```
Comment (Oleksandr-Tymoshenko) [inline src/main/java/com/champions/carsharingservice/controller/CarController.java:42]: you`ve forgotten about pageable

File: src/main/java/com/champions/carsharingservice/mapper/CarMapper.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+@Mapper(config = MapperConfig.class)
+public interface CarMapper {
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "deleted", ignore = true)
+    Car toEntity(CreateCarRequestDto request);
+
+    CarDto toDto(Car car);
+
+    void updateCar(CreateCarRequestDto dto, @MappingTarget Car car);
+
```
Comment (Oleksandr-Tymoshenko) [inline src/main/java/com/champions/carsharingservice/mapper/CarMapper.java:20]: remove blank line

**Review by Oleksandr-Tymoshenko — APPROVED** [review]
Good, but pay attention to comments

**Review by TetianaDenchyk — APPROVED** [review]
I approve but you should add pageable in future

**Pull request merged**